source/bringauto/common_utils/ProtobufUtils.cpp

```diff
@@ -97,4 +96,12 @@ ExternalProtocol::ExternalClient ProtobufUtils::createExternalClientCommandRespo
 	return externalMessage;
 }
 
-}
\ No newline at end of file
+void ProtobufUtils::copyStatusToBuffer(const InternalProtocol::DeviceStatus &status, bringauto::modules::Buffer &buffer) {
+	std::memcpy(buffer.getStructBuffer().data, status.statusdata().c_str(), status.statusdata().size());
```

Comment:
What if buffer is empty? What if data is nullptr?

Reply:
+ check the statusdata.size() is lett than equal to struct buffer. Same fo function below pls.

Reply:
added check if buffer is initialized